packages/hint-summary-display/src/hint.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview check for CSS rules that targets a summary tag and changes display to anything other than list-item
+ */
+
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint } from 'hint/dist/src/lib/types';
+import { StyleEvents, StyleParse } from '@hint/parser-css';
+import meta from './meta';
+import selectorParser = require('postcss-selector-parser');
+
+const parser = selectorParser();
+const requiredAttribute = 'display';
+const requiredValue = 'list-item';
+
+export default class ValidateSummaryHint implements IHint {
+    public static readonly meta = meta;
+
+    public constructor(context: HintContext<StyleEvents>) {
+
+        const validateStyle = ({ ast, resource }: StyleParse) => {
+
+            ast.walkRules((rule) => {
+
+                if (!rule.selector.includes('summary')) {
+                    return;
+                }
+
+                const validSelectors = parser.astSync(rule.selector).nodes.filter((node) => { // eslint-disable-line no-sync
+                    if (!('nodes' in node)) {
+                        return false;
+                    }
+
+                    const { nodes } = node;
+
+                    // Reverse loop because if we find combinator before summary tag then it does not alter summary and we can break out
```

Comment:
NIT: I don't think a comment is necessary for iterating in reverse. Usage is pretty self-explanatory.

Reply:
FWIW I specifically asked for this comment to clarify why iterating in reverse allows for an early exit of the loop.